src/main/java/com/thebois/models/world/World.java

```diff
@@ -56,36 +45,65 @@ public World(final int worldSize, final int pawnCount) {
      * @param colony    The Colony that should be added to the world.
      */
     public World(final int worldSize, final Colony colony) {
-        pawnCount = colony.getBeings().size();
+        this.worldSize = worldSize;
+
         terrainMatrix = new ITerrain[worldSize][worldSize];
         for (int y = 0; y < worldSize; y++) {
             for (int x = 0; x < worldSize; x++) {
                 terrainMatrix[y][x] = new Grass(x, y);
             }
         }
         // Structures
+        // noinspection unchecked
         structureMatrix = new Optional[worldSize][worldSize];
         for (int y = 0; y < structureMatrix.length; y++) {
             for (int x = 0; x < structureMatrix[y].length; x++) {
                 structureMatrix[y][x] = Optional.empty();
             }
         }
+
+        canonicalMatrix = new ITile[worldSize][worldSize];
+        repopulateCanonicalMatrix();
```

Comment:
Yeah, repopulating it from being empty. 

It was called just populate before I had to call it when new structures are placed. Calling `populateCanonicalMatrix` every time a new structure is placed seemed even worse than calling `repopulateCanonicalMatrix` once in the constructor, no?

Reply:
if you repopulate it you also populate it, correct?

Reply:
Sounds a bit weird to populate every time a structure is placed no?

Reply:
>Sounds a bit weird to populate every time a structure is placed no?

You're contradicting yourself, but yeah I agree with this, that's why I called it `repopulate` since it's repopulating the matrix every time a structure is placed.

I'll just rename it to `calculateCanonicalMatrix`.